File: java/src/main/java/svix/Svix.java
package svix;

import openapi.svix.ApiClient;
import openapi.svix.Configuration;
import openapi.svix.auth.*;

public class Svix {
	private final Application application;
	private final Authentication authentication;
	private final Endpoint endpoint;
	private final EventType eventType;
	private final Message message;
	private final MessageAttempt messageAttempt;

	public Svix(String token) {
		this(token, new SvixOptions());
	}

	public Svix(String token, SvixOptions options) {
		ApiClient apiClient = Configuration.getDefaultApiClient();
		apiClient.setBasePath(options.getUrl());

		HttpBearerAuth HTTPBearer = (HttpBearerAuth) apiClient.getAuthentication("HTTPBearer");
		HTTPBearer.setBearerToken(token);

		Configuration.setDefaultApiClient(apiClient);

		application = new Application();
		authentication = new Authentication();
		endpoint = new Endpoint();
		eventType = new EventType();
		message = new Message();
		messageAttempt = new MessageAttempt();
	}

	public Application getApplication() {
		return application;
	}

	public Authentication getAuthentication() {
		return authentication;
	}

	public Endpoint getEndpoint() {
		return endpoint;
	}

	public EventType getEventType() {
		return eventType;
	}

	public Message getMessage() {
		return message;
	}

	public MessageAttempt getMessageAttempt() {
		return messageAttempt;
	}
}

File: kotlin/lib/src/main/kotlin/Svix.kt
package com.svix.kotlin

import com.svix.kotlin.internal.infrastructure.ApiClient
//import com.svix.kotlin.internal.DefaultApi
//import com.svix.kotlin.internal.auth.HttpBearerAuth
// Add svix option
class Svix(token: String, svixOptions: SvixOptions = SvixOptions()) {
	companion object {
		val VERSION = "0.28.0"
	}

	val application: Application
	val authentication: Authentication
	val endpoint: Endpoint
	val eventType: EventType
	val message: Message
	val messageAttempt: MessageAttempt

	init {
		ApiClient.accessToken = token
		
		application = Application(svixOptions.debugUrl)
		authentication = Authentication(svixOptions.debugUrl)
		endpoint = Endpoint(svixOptions.debugUrl)
		eventType = EventType(svixOptions.debugUrl)
		message = Message(svixOptions.debugUrl)
		messageAttempt = MessageAttempt(svixOptions.debugUrl)
	}
}

